tests/testthat/test-bayes.R

```diff
@@ -534,3 +536,54 @@ test_that("missing performance values", {
       )
   })
 })
+
+# ------------------------------------------------------------------------------
+test_that("tune_bayes() output for `iter` edge cases (#721)", {
+  # for `iter = 0`, ought to match `tune_grid()`
+  boots <- rsample::bootstraps(mtcars)
+  wf <-
+    workflows::workflow(
+      mpg ~ .,
+      parsnip::linear_reg(penalty = tune(), engine = "glmnet")
+    )
+
+  ctrl_bayes <- control_bayes(seed = 1)
+
+  set.seed(1)
+  res_bayes <- tune_bayes(wf, boots, iter = 0, initial = 10,
+                          control = ctrl_bayes)
+
+  set.seed(1)
+  res_grid <- tune_grid(wf, boots)
+
+  expect_equal(
+    collect_metrics(res_bayes) %>% dplyr::select(-.iter),
+    collect_metrics(res_grid)
+  )
```

Comment:
The changes related to RNG are all related to this goal. I think `tune_bayes(..., iter = 0)` ought to match `tune_grid(...)`, but some function calls occurring before `grid_latin_hypercube()` altered the state of RNG, resulting in different initial grids even with the same seed.